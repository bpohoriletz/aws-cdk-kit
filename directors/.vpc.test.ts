import VpcDirector from "./vpc"
import SmallVpcBuilder from "../ec2/small-vpc-builder"

describe("new VpcDirector()", () => {
  let instance: VpcDirector;

  beforeEach(() => {
    instance = new VpcDirector(["stack"], SmallVpcBuilder);
  });

  test("construct vpc for pet projects", () => {
    expect(instance.constructVpc()).toMatchSnapshot();
  });
})
