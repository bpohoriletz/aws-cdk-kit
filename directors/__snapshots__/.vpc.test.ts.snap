// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`new VpcDirector() construct vpc for pet projects 1`] = `
VpcProduct {
  "ipAddressesSet": Cidr {
    "cidrBlock": "10.255.0.0/16",
    "networkBuilder": NetworkBuilder {
      "networkCidr": CidrBlock {
        "cidr": "10.255.0.0/16",
        "mask": 16,
        "networkAddress": 184549375,
        "networkSize": 65536,
      },
      "nextAvailableIp": 184483840,
      "subnetCidrs": [],
    },
  },
  "maxAzs": 1,
  "natGatewayProvider": NatInstanceProviderV2 {
    "gateways": PrefSet {
      "map": {},
      "next": 0,
      "vals": [],
    },
    "props": {
      "instanceType": InstanceType {
        "instanceTypeIdentifier": "t2.nano",
      },
    },
  },
  "natGatewaySubnets": {
    "subnetType": "Public",
  },
  "subnetConfiguration": [
    {
      "cidrMask": 19,
      "name": "Public",
      "subnetType": "Public",
    },
    {
      "cidrMask": 19,
      "name": "Private",
      "subnetType": "Private",
    },
  ],
}
`;
